feat: improve Thunder's menu GQLS for better accessibility and SEO

IT-Cru opened the pull request:
To improve accessibility and SEO of menus in Thunder GQLS related projects it would be good to add following elements from Drupal core to Thunder's menu GQLS schema for each menu item.

- [x] Description
- [x] Expanded
- [ ] Attributes (not included in this PR yet!)

### Discussion with community

Attributes are not included, because here some general key/value Attribute type should be added to Thunder's GQLS and maybe GraphQL's data producer for menu_link_attribute should be checked, if this could not also provide an array of attributes instead of using it per attribute.

These attributes could be useful for other contrib modules like [menu_link_attributes](https://www.drupal.org/project/menu_link_attributes) to get its data via GraphQL query.

### What is missing

- [ ] Testing data for menus needs to be extended to validate data for new GraphQL fields.
- [ ] Update of Headless API docs needs to be updated in [menu section](https://thunder.github.io/developer-guide/headless.html#menu).

Make sure these boxes are checked before submitting your pull request - thank you!

- [ ] All coding styles are fulfilled. ([How to check for cs issues?](https://github.com/BurdaMagazinOrg/thunder-dev-tools/blob/master/README.md#code-style-guidelines))
- [ ] All tests are running locally. ([How to run the test?](https://github.com/thunder/thunder-distribution/blob/8.x-4.x/docs/development.md#how-to-run-the-tests))
- [ ] Necessary update hooks are provided.
- [ ] User roles have correct access for new introduced permission.
- [ ] Every thunder module has a README.md in its root. Follow [this guidelines](https://www.drupal.org/node/2181737), but we don't need every topic.
- [ ] Code is covered with well-balanced amount of inline comments.
- [ ] New features or changes are documented.

If you are really awesome, then your feature is covered by additional tests. Well done!

dbosen committed:
News article

chrfritsch committed:
Add reference field to ParagraphVideo type

* feat(thunder_paragraphs): add src field to ParagraphVideo type
feat(thunder_gqls): add MediaSourceField data producer plugin for GraphQL

feat(thunder_gqls): add ThunderMediaProvider data producer plugin
fix(thunder_gqls): fix MediaVideo src field resolver to use media_source_field data producer plugin

feat(thunder_gqls): add src field resolver for ParagraphVideo GraphQL type
test(thunder_gqls): update paragraphs.query.graphql and paragraphs.response.json to include src field for ParagraphVideo type

* Fix

* Fix

* Rename src to reference

---------

Co-authored-by: Christian Fritsch <christian.fritsch@burda.com>

chrfritsch committed:
Allow multiple media types in entity browser view

* Allow multiple media types in entity browser view

* Add update hook

* Fix cs

* Better description

---------

Co-authored-by: Christian Fritsch <christian.fritsch@burda.com>

chrfritsch committed:
Add xymatic integration

* feat(composer.json): add drupal/xymatic dependency
feat(thunder_xymatic): add core.entity_form_display.media.xymatic.bulk_edit.yml configuration file

feat(thunder_xymatic): add default entity form display for media bundle xymatic

feat(thunder_xymatic): add media form display and field for xymatic bundle

feat(thunder_xymatic): add media type and field storage for Xymatic integration

* Add install hook

* Fix cs

* Rm

---------

Co-authored-by: Christian Fritsch <christian.fritsch@burda.com>

chrfritsch committed:
Release 6.5.4

* Release 6.5.4

* Fix

* Add news article

---------

Co-authored-by: Christian Fritsch <christian.fritsch@burda.com>

IT-Cru committed:
feat: improve Thunder's menu GQLS for better accessibility and SEO

IT-Cru committed:
Add configurable quick links to ThunderParagraphsTestTrait

dbosen reviewed (changes requested):
Implementation looks good. 
Please extend the menu.query test to include the new fields.

chrfritsch committed:
feat(thunder_xymatic): add default and paragraph_preview entity view displays for media.xymatic bundle (#704)

feat(thunder_xymatic): add media.xymatic.thumbnail view display configuration file

Co-authored-by: Christian Fritsch <christian.fritsch@burda.com>

dbosen committed:
Merge branch '6.5.x' into feature/improve-thunder-menu-gqls

dbosen committed:
Fix code style issues

chrfritsch committed:
Disable gpu and shared memory for webdriver

chrfritsch committed:
Add description to xymatic

* Add caption and description to xymatic

* Remove caption

---------

Co-authored-by: Christian Fritsch <christian.fritsch@burda.com>

chrfritsch committed:
Add video and xymatic type

* Add video and xymatic type

* Move from entity to metaData

* Update thunder_paragraphs.base.graphqls

---------

Co-authored-by: Christian Fritsch <christian.fritsch@burda.com>

chrfritsch committed:
fix: tests are indicating a success, but never ran

* Test the testing

* try using selenium url without env vars

* escape args

* debug

* more args

* use localhost

* add back env

* add back env

* no para

* set env vars

* no env

* escape

* show env

* trigger

* fix args

* show env

* move

* hardcore chromedriver url

* Set vars

* remove escape

* hardcore again

* remove filter

* Update tests.yml

* Back to normal

---------

Co-authored-by: Christian Fritsch <christian.fritsch@burda.com>
Co-authored-by: Volker Killesreiter <killesre@gmail.com>

dbosen committed:
Issue #3361796: Release Thunder 6.5.5

ol0lll committed:
Use drupalci chromedriver for testing.

dbosen committed:
Fix tests

- remove instagram test 
- improve mysql deadlocks
- fix metatag tests

dbosen committed:
Fix order of use statements.

dbosen committed:
Add decorated type resolver

dbosen committed:
Issue #3366947 by chr.fritsch: Release Thunder 6.5.6

dbosen committed:
Composer project version 3.x

ol0lll committed:
Release 6.5.7

Remove patch and unpin media_entity_pinterest.

dbosen committed:
Merge branch '6.5.x' into feature/improve-thunder-menu-gqls

dbosen committed:
Merge remote-tracking branch 'origin/7.2.x' into feature/improve-thunder-menu-gqls

dbosen committed:
cleanup

dbosen committed:
Merge branch '7.2.x' into feature/improve-thunder-menu-gqls

dbosen committed:
Merge remote-tracking branch 'origin/7.3.x' into feature/improve-thunder-menu-gqls

chrfritsch committed:
Merge branch '8.0.x' into feature/improve-thunder-menu-gqls
